src/main/java/techreborn/enet/ElectricNetwork.java

```diff
@@ -0,0 +1,238 @@
+	}
+
+	public void tick() {
+		if (isEmpty()) {
+			return;
+		}
+
+		if (isDirty) {
+			TechReborn.LOGGER.debug("Electric Network {} is dirty, checking for network and connection changes", this);
+			walkNetwork();
+
+			isDirty = false;
+		}
+
+		distributePower();
+	}
+
+	public boolean isEnergized() {
+		return isEnergized;
+	}
+
+	private void walkNetwork() {
+		// If the network is empty, why bother?
+		if (cableBlockEntities.isEmpty()) {
+			return;
+		}
+
+		TechReborn.LOGGER.debug("Walking the Electric Network to detect for changes in neighbors...");
+
+		// Keep track of the nodes we've seen and visited
+		HashSet<CableBlockEntity> seenNodes = new HashSet<>();
+		HashSet<CableBlockEntity> visitedNodes = new HashSet<>();
+
+		// "See" the first node in the list
+		CableBlockEntity firstNode = cableBlockEntities.iterator().next();
+		TechReborn.LOGGER.debug("First node in the walk: {}", firstNode);
+		seenNodes.addAll(visitCableNode(firstNode, visitedNodes));
+
+		// "remainingNodesToVisit" is our seen nodes, excluding the ones we've visited
+		HashSet<CableBlockEntity> remainingNodesToVisit = new HashSet<>(seenNodes);
+
+		while (!remainingNodesToVisit.isEmpty()) {
+			for (CableBlockEntity nodeToVisit : remainingNodesToVisit) {
+				seenNodes.addAll(visitCableNode(nodeToVisit, visitedNodes));
+			}
+
+			remainingNodesToVisit.clear();
+			remainingNodesToVisit.addAll(seenNodes);
+			remainingNodesToVisit.removeAll(visitedNodes);
+		}
+
+		for (CableBlockEntity nodeConnected : seenNodes) {
+			if (nodeConnected.getElectricNetwork() != this) {
+				TechReborn.LOGGER.debug(
+						"Block {} network change: {} -> {}",
+						nodeConnected.getPos().asLong(),
+						nodeConnected.getElectricNetwork(),
+						this);
+
+				nodeConnected.setElectricNetwork(this);
+			}
+		}
+
+		TechReborn.LOGGER.debug(
+				"Starting purge of disconnected entities. Have {}, seen {}",
+				cableBlockEntities.size(),
+				seenNodes.size());
+
+		cableBlockEntities
+				.stream()
+				.filter(node -> !seenNodes.contains(node))
+				.collect(Collectors.toList())
+				.forEach(node -> {
+					if (node != null) {
+						TechReborn.LOGGER.debug(
+								"Block {} being ejected from the network {}",
+								node,
+								this);
+
+						node.setElectricNetwork(null);
+					}
+				});
+
+		TechReborn.LOGGER.debug("Finished walking network for cable connection changes.");
+	}
+
+	private HashSet<CableBlockEntity> visitCableNode(CableBlockEntity currentEntity, HashSet<CableBlockEntity> visitedNodes) {
+		visitedNodes.add(currentEntity);
+
+		HashSet<CableBlockEntity> seenNodes = new HashSet<>();
+
+		if (currentEntity != null) {
+			TechReborn.LOGGER.debug(
+					"Visiting node {} on behalf of {} ({} visited total)",
+					currentEntity.getPos().asLong(),
+					this,
+					visitedNodes.size());
+
+			seenNodes.add(currentEntity);
+
+			for (Direction face : Direction.values()) {
+				BlockPos blockToCheckPos = currentEntity.getPos().offset(face);
```

Comment:
Did you have a chance to test how this will work when part of network is in unloaded chunk?

Reply:
Yes, you need to be very careful here as it if a cable is on the edge of a chunk boarder its going to start loading the adjacent chunks causing all sorts of issues.

There is a way to check if a pos is in a loaded chunk but, id guess there is some more work to do around this.